### ISSUE TITLE
Up/down buttons very difficult to select in Firefox, e.g. when trying to sell food items

### ISSUE THREAD

**Issue by shanaqui** [HabitRPG/habitica#13252]
![image](https://user-images.githubusercontent.com/8144640/117436333-c98cd480-af26-11eb-8e60-9cfc62085acd.png)

Reported by Vikte and confirmed by me.

"[Market] In Firefox, up-down buttons when selling items only react when you hover the mouse EXACTLY over their right border, and don't react when mouse is right above the button. In Edge it works fine."

**Comment by fish-dango** [comment]
Hello! I would like to try this issue out if that is okay?

**Comment by Alys** [comment]
@fish-dango Sure, thank you! Post here if you have any questions or run into problems!

**Comment by shanaqui** [comment]
@fish-dango Hi there! Are you still working on this one? If not, or if you don't reply within a week, we're going to mark this as "help wanted" again to make sure people know it's available to work on. Thanks!

**Comment by jasonmishi** [comment]
![image](https://user-images.githubusercontent.com/55836477/135017053-6362e61e-f6bd-4665-8269-29b779f4c52d.png)
This is resolved in my end but that might be due to my browser version.
![image](https://user-images.githubusercontent.com/55836477/135017191-d7aa530e-e092-4a51-ba0f-29ac15547d5a.png)

**Comment by SabreCat** [comment]
This has definitely been improved since the ticket was written, but I feel like there's still improvement to be made here--the arrows are not very easy targets even still, and there's plenty of space in the control for them to be larger. @Tressley I'm not seeing a standard styleguide for numeric spinner controls on Zeplin--how would you like to handle this?

**Comment by Tressley** [comment]
It's a number input, so the user does have the ability to type in the amount in addition to the step controls. However, it sounds like lower numbers might be more common and users find it easier to click a few times instead of typing in the amount. In that case, we could definitely mess with the styling (remove padding and define height via `line-height`), but I don't think that'll give us the level of control we're looking for.

I think the solution would be to introduce a new text-field increment stepper component using a solution like this: http://jsfiddle.net/74xa8Ler/

This allows us to pull the step controls out of the input and design them a bit more while also having the text entry method. I'll throw a design together. 👍🏻

**Comment by SabreCat** [comment]
@Tressley I know there's been plenty else going on; should we leave this on hold, or do you feel like knocking this out yet?

**Comment by Tressley** [comment]
Here they are! https://zpl.io/bL3Ej67

**Comment by ReCore-sys** [comment]
Hey! First time doing anything here so it might take me a day or so to get a hang of the codebase but can I take a shot at this one?
Also, any specific ideas for what design we want for this? And do we know what version of firefox is not working right?

**Comment by CuriousMagpie** [comment]
Hey there @ReCore-sys! Here's the link to the design--I think @Tressley will need to give you access to Zeplin for it to work for you: https://github.com/HabitRPG/habitica/issues/13252#issuecomment-966499227

I'll go ahead and assign this to you. Thanks for your help!

**Comment by Tressley** [comment]
@ReCore-sys -- Can you provide me with an email address to invite to our Zeplin workspace?

**Comment by ReCore-sys** [comment]
Yeah sure. kjshore2004@gmail.com

**Comment by iswarmondal** [comment]
### **Here is the code to solve the problem**
**Just copy the code and test it out using an HTML file**
@ReCore-sys @CuriousMagpie 

```
<!DOCTYPE html>
<html>
<head>
	<style>
		input[type="number"] {
		-webkit-appearance: textfield;
			-moz-appearance: textfield;
				appearance: textfield;
		}
		input[type=number]::-webkit-inner-spin-button, 
		input[type=number]::-webkit-outer-spin-button { 
			-webkit-appearance: none;
		}
	</style>
</head>
<body>

Number: 
<button onclick="myFunction2()">-</button>
<input type="number" id="myNumber">
<button onclick="myFunction()">+</button>

<p><strong>Note:</strong> The stepUp() method is not supported by Internet Explorer.</p>

<script>
function myFunction() {
  document.getElementById("myNumber").stepUp(1);
}
function myFunction2() {
  let value = document.getElementById("myNumber").value;
  if(value>0){
  document.getElementById("myNumber").stepDown(1);
  }
}
</script>

</body>
</html>
```

**Comment by CuriousMagpie** [comment]
@ReCore-sys are you still working on this?

**Comment by ReCore-sys** [comment]
Nah. The other guy managed to come up with a solution and I haven't played around with the testing environment enough to implement it, so not much point.

**Comment by CuriousMagpie** [comment]
Fair enough, @ReCore-sys!

@iswarmondal Please open a PR with your fix. Thanks!

**Comment by CuriousMagpie** [comment]
This solution does not work with our codebase, therefore we're opening it up to contributors again.

**Comment by KTKreature** [comment]
Could I try to solve this?

**Comment by CuriousMagpie** [comment]
@KTKreature Sure thing! Just for clarity's sake, I am currently working on a similar issue--feel free to look at [my progress here](https://github.com/CuriousMagpie/habitica/blob/gifting-modal-design/website/client/src/components/payments/sendGiftModal.vue) and take whatever you think is useful. We would like to have this part be a component so we can easily deploy it in other parts of the site.

**Comment by CuriousMagpie** [comment]
@KTKreature Hey there! Have you done any work on this? I have come up with an implementation and have been given the task of turning it into a component and deploying it across the site, so I'm going to assign this to myself and unassign you. Feel free to look at our other open issues!

**Comment by TheHackerWitch-Official** [comment]
Hi! I'm interested in contributing to Habitica. Is this issue being worked on?

**Comment by Tressley** [comment]
@ch4kra -- Thank you for your interest in contributing to Habitica! Yes, the componentization is still being worked on by @CuriousMagpie. Please feel free to have a look at our other open issues.

**Comment by SabreCat** [comment]
Fixed in 2448f401f2d0a1cb87c935bc945cc0ef5fd23d1c
